### PR TITLE
Add mountainsort5 sorter

### PR DESCRIPTION
**Pull request by magland**
Adding a new version of mountainsort spike sorter. See https://github.com/magland/mountainsort5

The structure of the sorting parameters is a bit more complicated than for mountainsort4 because the algorithm is available in three different schemes (scheme1, scheme2, scheme3) -- see the above link. Each scheme has its own set of parameters. So for the SpikeInterface wrapper, I flattened the structure... see mountainsort5.py. Depending on which scheme is selected, different parameters will take effect (and others will be ignored).

### CODE REVIEW

File: src/spikeinterface/sorters/external/mountainsort4.py [hunk 1]
```diff
@@ -105,7 +105,7 @@ def _run_from_folder(cls, sorter_output_folder, params, verbose):
         if p['whiten']:
             if verbose:
                 print('whitening')
-            recording = whiten(recording=recording)
+            recording = whiten(recording=recording, dtype='float32')
```
Comment (alejoe91) [inline src/spikeinterface/sorters/external/mountainsort4.py:108]: this should address this issue too #1512

File: src/spikeinterface/sorters/external/tests/test_singularity_containers.py [hunk 2]
```diff
@@ -66,6 +66,12 @@ def test_mountainsort4(run_kwargs):
     print(sorting)
 
 
+def test_mountainsort5(run_kwargs):
+    sorting = ss.run_sorter("mountainsort5", output_folder=cache_folder / "mountainsort5", **run_kwargs)
+    print("resulting sorting")
+    print(sorting)
```
Comment (alejoe91) [inline src/spikeinterface/sorters/external/tests/test_singularity_containers.py:72]: @magland can you make a PR with a Dockerfile here: https://github.com/SpikeInterface/spikeinterface-dockerfiles ?

So we can run it through this singularity test :)

**Review by alejoe91 — APPROVED** [review]
Looks good to me! 

It would be great if you can also make an associated Docker image (which should be as straightforward as [mountainsort4](https://github.com/SpikeInterface/spikeinterface-dockerfiles/blob/main/mountainsort4/Dockerfile))

**Comment by magland** [comment]
@alejoe91 before you merge...

I'm wondering about what will happen if the mountainsort5 parameters or api changes. This would break the SI wrapper. So maybe we should have SI depend on a narrow version range for mountainsort5? Then if I update to a new major version, then I'd make a new PR? What do you think?

**Comment by alejoe91** [comment]
> @alejoe91 before you merge...
> 
> I'm wondering about what will happen if the mountainsort5 parameters or api changes. This would break the SI wrapper. So maybe we should have SI depend on a narrow version range for mountainsort5? Then if I update to a new major version, then I'd make a new PR? What do you think?

Yeah I think that's ok! You can check the version in the [is_installed()](https://github.com/magland/spikeinterface/blob/mountainsort5/src/spikeinterface/sorters/external/mountainsort5.py#L76-L82) and maybe warn the user if they need to downgrade mountainsort5?

I think this is the exact reason why we should have a working Dockerfile with the current version :)

**Comment by magland** [comment]
@alejoe91 I added the check for mountainsort5 version 0.1.x
This requires the semantic_version package which is very standard (probably already installed on user's system). Should we make this explicit in the pyproject.toml, or just let it be?

https://github.com/magland/spikeinterface/blob/42f3c0da8b0c7690ea09a782b6506d8f7cbd6b9d/src/spikeinterface/sorters/external/mountainsort5.py#L76-L87

Next I will work on the singularity container.

**Comment by magland** [comment]
@alejoe91 I have made a PR to spikeinterface-dockerfiles

**Comment by alejoe91** [comment]
> @alejoe91 I added the check for mountainsort5 version 0.1.x This requires the semantic_version package which is very standard (probably already installed on user's system). Should we make this explicit in the pyproject.toml, or just let it be?
> 
> https://github.com/magland/spikeinterface/blob/42f3c0da8b0c7690ea09a782b6506d8f7cbd6b9d/src/spikeinterface/sorters/external/mountainsort5.py#L76-L87
> 
> Next I will work on the singularity container.

We use `from packaging.version import parse` in other places. Could you use that one?
```
from packaging.version import parse

if parse(mountainsort5.__version__) > parse("0.1"):
       print(f'WARNING: This version of SpikeInterface expects Mountainsort5 version 0.1.x. You have version {mountainsort5.__version__}')
       HAVE_MS5 = False
```

**Comment by magland** [comment]
> We use `from packaging.version import parse` in other places. Could you use that one?
> 
> ```
> from packaging.version import parse
> 
> if parse(mountainsort5.__version__) > parse("0.1"):
>        print(f'WARNING: This version of SpikeInterface expects Mountainsort5 version 0.1.x. You have version {mountainsort5.__version__}')
>        HAVE_MS5 = False
> ```

Okay I have made the change.

**Review by alejoe91 — APPROVED** [review]

**Pull request merged**